largest_product_array/largest_product.py

```diff
@@ -0,0 +1,16 @@
+
+
+def largest(input_list):
+    """
+    function accepts a 2d list of paired ints and returns the largest product of adjacent ints
+    """
+    if not input_list:
+        return 'empty list used'
+    largest_prod = 0
+    product = 0
+    for index in input_list:
+        product = index[0] * index[1]
+        if product > largest_prod:
+            largest_prod = product
+    return largest_prod
```

Comment:
This only checks for products of number next to each other horizontally. You also need to check for values next to each other vertically.
```
[[3, 4], [5, 8], [1, 2]]
```
is equivalent to the matrix
```
3  4
5  8
1  2
```
So you need to multiply `3 * 4` as well as `3 * 5` in order to find the max.
Also your solution should work for any size matrix, not just 2xn.